### DISCUSSION TITLE
Issue with router.push

### DISCUSSION THREAD

**Discussion by amanSethSmava** [vercel/next.js#37923] [Help]
I am using next along with react 18.
On using push, url of my app gets changed by app is stuck in a loop as the new page keeps on getting rendered. 

This issue was not present when I was using react 17. 

I am using push something like

```
push({
        pathname: `${routeName}`,
        hash: `${hashValue}`,
      })

```
Is this bug known or anyone has a similar issue

**Comment by JeonJaewon** [comment]
Did you call `push` function of `useRouter` inside `useEffect`?
There's a infinite loop [issue](https://github.com/vercel/next.js/issues/18127).

**Reply by amanSethSmava** [reply]
Not use effect, inside useCallback

**Reply by amanSethSmava** [reply]
even with a basic function, it's landing in an infinite loop
